client/asset/btc/spv_wrapper.go

```diff
@@ -1745,12 +1762,12 @@ func (w *spvWallet) getWalletTransaction(txHash *chainhash.Hash) (*GetTransactio
 	*/
 }
 
-func confirms(txHeight, curHeight int32) int32 {
+func confirms(txHeight, curHeight int32) uint32 {
 	switch {
-	case txHeight == -1, txHeight > curHeight:
-		return 0
+	case curHeight >= txHeight:
+		return uint32(curHeight - txHeight + 1) // positive numbers here, no overflow
 	default:
-		return curHeight - txHeight + 1
+		return 0 // undefined for all other cases
```

Comment:
One thing we're missing with these changes to `confirms` is special handling when `txHeight` is -1.  When this helper is used with `txHeight, _ := w.cl.GetBlockHeight(blockHash)` as in our `getBlockHeader` method, I don't think we'd ever get -1, **however** this helper with `wtxmgr.TxDetails` types like `txDetails.Block.Height` where -1 is expected in caess and we don't want to change any behavior there e.g. `getTxOut`.

Note that this `confirms` helper is lifted directly from dcrwallet:  https://github.com/decred/dcrwallet/blob/0aeb02cde83ebb552b802b4eebcc151fe371325c/wallet/wallet.go#L4446-L4456

Let's revert the changes to this method please.